layers/meta-balena-raspberrypi/recipes-core/initrdscripts/files/kexec_pi4_fwgpio

```diff
@@ -0,0 +1,56 @@
+#!/bin/sh
+
+# shellcheck disable=SC1091
+. /usr/libexec/os-helpers-logging
+
+MMC_DEV="fe300000.mmcnr"
+MMC_SYSFS_DIR="/sys/bus/platform/drivers/mmc-bcm2835"
+GPIO_SYSFS_DIR="/sys/class/gpio"
+
+FW_GPIO_BASE="504"
```

Comment:
I've really tried to figure out where the "504" comes from, but I just couldn't. I do not think it is explicitly defined anywhere in the device tree, my best guess at this moment is that somewhere there is a cap for 512 GPIOs and the 8 firmware pins map to the end, so 512 - 8 = 504, but I was not able to prove that either.

It is always 504 on the CM4, Pi4 and Pi400 and since the script is only specific to that family, I did not want to spend too much time on a generic solution.